packages/core/src/common/uri.ts

```diff
@@ -199,8 +199,17 @@ export default class URI {
         return this.codeUri.toString(skipEncoding);
     }
 
+    isEqual(uri: URI, caseSensitive: boolean = true): boolean {
+        if (!this.hasSameSchemeAuthority(uri)) {
+            return false;
+        }
+
+        return (!caseSensitive && this.toString().toLowerCase() === uri.toString().toLowerCase())
+            || (this.toString() === uri.toString());
+    }
```

Comment:
For simplicity, and readability:

```suggestion
    isEqual(uri: URI, caseSensitive: boolean = true): boolean {
        if (!this.hasSameSchemeAuthority(uri)) {
            return false;
        }
        
        return !caseSensitive 
        ? this.toString().toLowerCase() === uri.toString().toLowerCase())
        : this.toString() === uri.toString();
   }
```